### PR TITLE
Fix scrollbar drag gestures for reversed scrollables

### LINKED ISSUE #81035: Scrollbar is not interactive in reversed ListView
## Steps to Reproduce

I cannot drag Scrollbar with cursor if ListView has `reverse: true` property (chrome and windows builds)

```dart
Scrollbar(
  isAlwaysShown: true,
  interactive: true,
  child: ListView.builder(
    itemCount: _nums.length,
    reverse: true,
    itemBuilder: itemBuilder,
  ),
),
```

**Expected results:** Scrollbar is draggable

**Actual results:** Scrollbar is not draggable

<details>
  <summary>Logs</summary>

flutter doctor -v
```
[√] Flutter (Channel beta, 2.2.0-10.1.pre, on Microsoft Windows [Version 10.0.19042.928], locale ru-RU)
    • Flutter version 2.2.0-10.1.pre at C:\flutter
    • Framework revision 0941968447 (8 days ago), 2021-04-15 12:01:02 -0700
    • Engine revision d2a2e93510
    • Dart version 2.13.0 (build 2.13.0-211.6.beta)

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\Vlad\AppData\Local\Android\Sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = C:\Users\Vlad\AppData\Local\Android
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Users\Vlad\AppData\Local\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.9.3)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.9.31129.286
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] IntelliJ IDEA Ultimate Edition (version 2020.1)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA 2020.1.1
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] VS Code (version 1.55.2)
    • VS Code at C:\Users\Vlad\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.21.0

[√] Connected device (4 available)
    • SM A505FM (mobile) • R58M310EWYE • android-arm64  • Android 11 (API 30)
    • Windows (desktop)  • windows     • windows-x64    • Microsoft Windows [Version 10.0.19042.928]
    • Chrome (web)       • chrome      • web-javascript • Google Chrome 90.0.4430.85
    • Edge (web)         • edge        • web-javascript • Microsoft Edge 89.0.774.77

• No issues found!
```

</details>

### PR DESCRIPTION
**Pull request by Piinks**
Fixes https://github.com/flutter/flutter/issues/81035

Drag gestures were not accounting for reversed scroll views. 🙃 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/widgets/scrollbar.dart [hunk 1]
```diff
@@ -1113,19 +1117,25 @@ class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProv
...
     _fadeoutTimer?.cancel();
     _fadeoutAnimationController.forward();
     switch (direction) {
-      case Axis.vertical:
+      case AxisDirection.down:
         _dragScrollbarAxisPosition = localPosition.dy;
         break;
-      case Axis.horizontal:
+      case AxisDirection.up:
+        _dragScrollbarAxisPosition = _flipAxis(localPosition.dy);
```
Comment (chunhtai) [inline packages/flutter/lib/src/widgets/scrollbar.dart:1131]: I suggest we do all delta calculation in _updateScrollPosition so we don't need to keep track of the axis direction in handleThumbPressStart and handleThumbPressUpdate. The _updateScrollPosition can take in an Offset.

Also, do we handle corner cases like the axis direction changes during a thumb press?

Comment (Piinks) [reply packages/flutter/lib/src/widgets/scrollbar.dart:1131]: I am not sure I follow. We still need to keep track of the position where the gesture started so when we get an updated position we can make the calculation. Which Offset do you want `_updateScrollPosition` to receive?

Comment (chunhtai) [reply packages/flutter/lib/src/widgets/scrollbar.dart:1131]: something like this
```dart
Offset _dragScrollbarAxisPosition;
void handleThumbPressStart(Offset localPosition) {
   _dragScrollbarAxisPosition = localPosition;
}

void handleThumbPressUpdate(Offset localPosition) {
  _updateScrollPosition(localPosition);
}

void _updateScrollPosition(Offset currentLocation) {
     double primaryDelta;
     switch (direction) {
         // calculate delta
     }
}

```

Comment (Piinks) [reply packages/flutter/lib/src/widgets/scrollbar.dart:1131]: That's what I thought. About to push an update. 😜

**Review by chunhtai — APPROVED** [review]
LGTM

**Pull request merged**